### ISSUE TITLE
How to use afl-clang/afl-clang-fast compiler the whole pregram?

### ISSUE THREAD

**Issue by hrshy0629** [travitch/whole-program-llvm#135]
Hi, sir. I use the WLLVM to compile the kernel and other programs (export LLVM_COMPILER=clang), which has obtained many good results in the past. However, I want to use the WLLVM to compile other programs( in this case, I want to use "export LLVM_COMPILER=afl-clang-fast" or "export LLVM_COMPILER=afl-clang" to compile the programs. The fl-clang-fast and afl-clang are the wrapper of the CLANG and implement the instrumentation in source code during the compilation process). But WLLVM doesn't work.
 
So, sir. Can you tell me about WLLVM principles or reference manuals, or do you have plans to complement this work? Do you have any suggestions or references if I want to do this?

**Comment by ianamason** [comment]
Why doesn't just wrapping afl-clang rather than clang work? The README.md is the manual.

**Comment by ianamason** [comment]
All wllvm does is call the wrapped compiler to generate the bitcode, then stores the path to the bitcode in a section in
the object.  I don't see why just wrapping afl-clang/afl-clang-fast won't give you the bitcode that afl-clang produces.

**Comment by hrshy0629** [comment]
Thanks, your first sentence makes me understand how WLLVM works. I know how to do it and need to change some source code and experiment with it.

**Comment by lijunqiang123** [comment]
> Thanks, your first sentence makes me understand how WLLVM works. I know how to do it and need to change some source code and experiment with it.

I also encountered this problem recently. Can you provide corresponding solutions? That would help me a lot. Thank you!

**Comment by hrshy0629** [comment]
In fact, I gave up compiling with wllvm, which was hard to follow.



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "travitch/whole-program-llvm"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年9月20日(星期三) 晚上7:26
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [travitch/whole-program-llvm] How to use afl-clang/afl-clang-fast compiler the whole pregram? (Issue #135)





  
Thanks, your first sentence makes me understand how WLLVM works. I know how to do it and need to change some source code and experiment with it.
  
I also encountered this problem recently. Can you provide corresponding solutions? That would help me a lot. Thank you!
